src/lib.rs

```diff
@@ -193,21 +228,26 @@ impl Wallet {
             .sync(BdkProgressHolder { progress_update }, max_address_param)
     }
 
-    fn get_new_address(&self) -> String {
-        self.get_wallet()
-            .get_address(AddressIndex::New)
-            .unwrap()
-            .address
-            .to_string()
+    fn get_address(&self, address_index: AddressIndex) -> Result<AddressInfo, BdkError> {
+        match self.get_wallet().get_address(bdk::wallet::AddressIndex::from(address_index)) {
+            Ok(address_info) => Ok(AddressInfo::from(address_info)),
+            Err(bdk_error) => Err(bdk_error),
+        }
```

Comment:
we can rewrite this with a [`Result::map`](https://doc.rust-lang.org/std/result/enum.Result.html#method.map) and an [`Into::into`](https://doc.rust-lang.org/std/convert/trait.Into.html):
```rust
self.get_wallet()
    .get_address(address_index.into())
    .map(AddressInfo::from)
```